pkg/occlient/occlient.go

```diff
@@ -910,6 +928,23 @@ func uniqueAppendOrOverwriteEnvVars(existingEnvs []corev1.EnvVar, envVars ...cor
 	return retVal
 }
 
+// deleteNonRequiredEnvVars deletes the passed env var from the list of passed env vars
+// Parameters:
+//	existingEnvs: Slice of existing env vars
+//	envTobeDeleted: The name of env var to be deleted
+// Returns:
+//	slice of env vars with delete reflected
+func deleteNonRequiredEnvVars(existingEnvs []corev1.EnvVar, envTobeDeleted string) []corev1.EnvVar {
```

Comment:
I found this function name pretty confusing. 
It just removes one item from `EnvVar` slice, but name uses the plural "Vars", also why it is "NonRequired" when there is an argument that clearly determines what should be deleted.

Another problem is that if someone calls this function and assigns its return value to different variable than what is used as the first argument it will lead to nasty bugs :-(

```go
newEnvs := deleteNonRequiredEnvVars(originEnvs, deleteThis)
```
`newEnvs`  will have corrected output, but `originEnvs` will be also modified, and it will be somehting that you would't expect.

Reply:
> I found this function name pretty confusing.
> It just removes one item from `EnvVar` slice, but name uses the plural "Vars", also why it is "NonRequired" when there is an argument that clearly determines what should be deleted.
> 
> Another problem is that if someone calls this function and assigns its return value to different variable than what is used as the first argument it will lead to nasty bugs :-(
> 
> ```go
> newEnvs := deleteNonRequiredEnvVars(originEnvs, deleteThis)
> ```
>

Ack 

> `newEnvs` will have corrected output, but `originEnvs` will be also modified, and it will be somehting that you would't expect.

No, not since my previous-to-previous commit anyway I amended a UT also to test this now

Reply:
no, this is still an issue with this implementation

Reply:
Aah my bad :( 
Now fixed it as tried also in https://play.golang.org/p/LwcxpaSbqkv